xesmf/frontend.py

```diff
@@ -1151,3 +1192,83 @@ def _format_xroutput(self, out, new_dims=None):
         out.coords[self._lat_out_name] = xr.DataArray(self._lat_out, dims=(self.geom_dim_name,))
         out.attrs['regrid_method'] = self.method
         return out
+
+    @staticmethod
+    def densify_polys(polys, max_length=1.0):
+        """
+        Takes in a list of polygons and densifies each of the polygons' segments by adding equidistant points
+        between the two endpoints of each segment such that the maximum length of a segment is max_length.
+        max_length is set to 1.0 degree in lon/lat space by default.
+
+        Parameters
+        ----------
+        polys : list, List of polygons to densify
+
+        max_length : float, Maximum desired length of a segment. The default is 1 degree in lat/lon space.
+                            Each side of the polygon is then split into equidistant segments of length max_length
+
+        Returns
+        -------
+        List of densified polygons
+
+        """
+        dense_polys = []  # Densified polys
+        for poly in polys:
+            # If multipoly, we densify each poly
+            if isinstance(poly, MultiPolygon):
+                list_polys_dense = []
+                list_polys = list(poly.geoms)
+                for ipoly in list_polys:
+                    if isinstance(ipoly.boundary, MultiLineString):  # Poly has holes
```

Comment:
Do you think we could reduce the number of lines by creating a function for this section and the "single poly" one ?